test/e2e/storage/persistent_volumes-local.go

```diff
@@ -252,6 +253,62 @@ var _ = SIGDescribe("PersistentVolumes-local [Feature:LocalPersistentVolumes] [S
 		})
 	})
 
+	Context("when pod using local volume with non-existant path", func() {
+		ep := &eventPatterns{
+			reason:  "FailedMount",
+			pattern: make([]string, 2)}
+		ep.pattern = append(ep.pattern, "MountVolume.SetUp failed")
+		ep.pattern = append(ep.pattern, "does not exist")
+
+		It("should not be able to mount", func() {
+			for _, testVolType := range LocalVolumeTypes {
+				By(fmt.Sprintf("local-volume-type: %s", testVolType))
+				testVol := &localTestVolume{
+					node:            config.node0,
+					hostDir:         "/non-existent/location/nowhere",
+					localVolumeType: testVolType,
+				}
+				By("Creating local PVC and PV")
+				createLocalPVCPV(config, testVol)
+				pod, err := createLocalPod(config, testVol)
+				Expect(err).To(HaveOccurred())
+				checkPodEvents(config, pod.Name, ep)
+			}
+		})
+	})
+
+	Context("when pod's node is different from PV's NodeAffinity", func() {
+
+		BeforeEach(func() {
+			if len(config.nodes.Items) < 2 {
+				framework.Skipf("Runs only when number of nodes >= 2")
+			}
+		})
+
+		ep := &eventPatterns{
+			reason:  "FailedScheduling",
+			pattern: make([]string, 2)}
+		ep.pattern = append(ep.pattern, "MatchNodeSelector")
+		ep.pattern = append(ep.pattern, "No nodes are available")
+		for _, testVolType := range LocalVolumeTypes {
+
+			It("should not be able to mount due to different NodeAffinity", func() {
+
+				testPodWithNodeName(config, testVolType, ep, config.nodes.Items[1].Name, makeLocalPodWithNodeAffinity)
+			})
+
+			It("should not be able to mount due to different NodeSelector", func() {
+
+				testPodWithNodeName(config, testVolType, ep, config.nodes.Items[1].Name, makeLocalPodWithNodeSelector)
+			})
+
+			It("should not be able to mount due to different NodeName", func() {
```

Comment:
I ran this test case on my cluster and it failed to find the FailedScheduling event.  My guess is that by setting NodeName, you completely bypass the scheduler.  The pod should still fail to start though, due to the FailedMount event.

Reply:
Could you send me the logs you got and also is there any way to bring multinode cluster and run multinode version of e2e tests?

Reply:
I'm not sure how to mimic multiple nodes in a local cluster setup.  I brought up a real GCE multinode cluster to run the tests.

This was the e2e test log:
```
• Failure [336.703 seconds]
[sig-storage] PersistentVolumes-local [Feature:LocalPersistentVolumes] [Serial]
/usr/local/google/home/msau/workspace/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/storage/framework.go:22
  when pod's node is different from PV's NodeAffinity
  /usr/local/google/home/msau/workspace/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/storage/persistent_volumes-local.go:310
    should not be able to mount due to different NodeName [It]
    /usr/local/google/home/msau/workspace/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/storage/persistent_volumes-local.go:308

    Expected
        <int>: 0
    not to equal
        <int>: 0

    /usr/local/google/home/msau/workspace/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/storage/persistent_volumes-local.go:405
```

Here were the pod events:
```
Sep  6 11:11:05.905: INFO: At 2017-09-06 11:06:05 -0700 PDT - event for security-context-0d27fa21-932e-11e7-b9d4-480fcf4a51ea: {kubelet e2e-test-msau-minion-group-fd2r} FailedMount: Storage node affinity check failed for volume "local-pvgzlcp" : NodeSelectorTerm [{Key:kubernetes.io/hostname Operator:In Values:[e2e-test-msau-minion-group-6sv7]}] does not match node labels
Sep  6 11:11:05.905: INFO: At 2017-09-06 11:08:08 -0700 PDT - event for security-context-0d27fa21-932e-11e7-b9d4-480fcf4a51ea: {kubelet e2e-test-msau-minion-group-fd2r} FailedMount: Unable to mount volumes for pod "security-context-0d27fa21-932e-11e7-b9d4-480fcf4a51ea_e2e-tests-persistent-local-volumes-test-jl278(0d24f71c-932e-11e7-9d5e-42010a800002)": timeout expired waiting for volumes to attach/mount for pod "e2e-tests-persistent-local-volumes-test-jl278"/"security-context-0d27fa21-932e-11e7-b9d4-480fcf4a51ea". list of unattached/unmounted volumes=[volume1]
```